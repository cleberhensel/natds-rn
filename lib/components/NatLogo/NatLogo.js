import React from 'react';
import PropTypes from 'prop-types';
import { load as loadSvg } from '../../assets/logo/svg';

export const NatLogo = ({ name, size }) => {
  const IconComponent = loadSvg(name);

  return (
    <IconComponent width={size} />
  );
};

NatLogo.propTypes = {
  name: PropTypes.string,
  size: PropTypes.number,
};

NatLogo.defaultProps = {
  name: 'logo-natura-vertical',
  size: 24,
};
