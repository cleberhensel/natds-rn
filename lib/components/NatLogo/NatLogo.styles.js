import styled from 'styled-components/native';

const imageStyles = {
  card: {
    height: 184,
    width: '100%',
  },
  circle: {
    borderRadius: 100,
    height: 48,
    width: 48,
  },
  default: {
  },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getImageStyles = getStylesFrom(imageStyles);

export const Image = styled.Image`
  width: ${({ type }) => getImageStyles(type).width};
  height: ${({ type }) => getImageStyles(type).height};
`;
