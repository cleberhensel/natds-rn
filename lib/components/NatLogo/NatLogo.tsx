import React from "react";
import { load as loadSvg } from "../../assets/logo/svg";
type NatLogoProps = {
  height?: number,
  name?: string,
  size?: number,
  width?: number
};
export const NatLogo: React.SFC<NatLogoProps> = ({
  name,
  size,
  width,
  height
}) => {
  const IconComponent = loadSvg(name);
  return (
    <IconComponent
      width={width > 0 ? width : size}
      height={height > 0 ? height : size}
    />
  );
};
NatLogo.defaultProps = {
  height: 0,
  name: "logo-natura-vertical",
  size: 0,
  width: 0
};
