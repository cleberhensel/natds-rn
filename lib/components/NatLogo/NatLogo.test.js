import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { NatLogo } from './NatLogo';

describe('NatLogo component default', () => {
  it('Should render NatLogo component ', () => {
    const button = renderer.create(
      <NatLogo />,
    ).toJSON();

    expect(button).toMatchSnapshot('NatLogo snapshot');
  });
});
