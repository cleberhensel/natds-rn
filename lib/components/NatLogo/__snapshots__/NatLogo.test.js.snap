// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatLogo component default Should render NatLogo component  1`] = `
<SvgMock
  height={0}
  width={0}
/>
`;
